tasks/libs/notify/utils.py

```diff
@@ -43,3 +44,30 @@ def get_ci_visibility_job_url(
         extra_args = ''.join([f'&{key}={value}' for key, value in extra_args.items()])
 
     return CI_VISIBILITY_JOB_URL.format(name=name, extra_flags=extra_flags, extra_args=extra_args)
+
+
+def should_notify():
+    """
+    Check if the pipeline should notify the channel: only for non-downstream pipelines, unless conductor triggered it
+    """
+    from tasks.libs.ciproviders.gitlab_api import get_pipeline
+
+    CONDUCTOR_ID = 8278
+    pipeline = get_pipeline(PROJECT_NAME, os.environ['CI_PIPELINE_ID'])
+    return (
+        os.environ['CI_PIPELINE_SOURCE'] != 'pipeline'
+        or os.environ['CI_PIPELINE_SOURCE'] == 'pipeline'
+        and pipeline.user['id'] == CONDUCTOR_ID
```

Comment:
Are we reasonably confident that the conductor id is not going to change over time?

Reply:
I suppose this is linked to the conductor user that triggers the pipeline. Even if I see no reason for it to change, I might think about a warning if this changes, indeed. 
An alternative would be to notified on `non codesync downstream`, but this is weak as well, as we can have manually triggered pipelines.

Reply:
I changed this to use the environment variables that are passed by the conductor nightly pipeline [here](https://github.com/DataDog/k8s-datadog-agent-ops/blob/main/.gitlab/nightly.yaml#L19-L20)